volatility/framework/symbols/windows/extensions/__init__.py

```diff
@@ -361,7 +362,9 @@ def is_valid(self) -> bool:
     def file_name_with_device(self) -> Union[str, interfaces.renderers.BaseAbsentValue]:
         name = renderers.UnreadableValue()  # type: Union[str, interfaces.renderers.BaseAbsentValue]
 
-        if self._context.layers[self.vol.layer_name].is_valid(self.DeviceObject):
+        # this pointer needs to be checked against native_layer_name because the object may
+        # be instantiated from a primary (virtual) layer or a memory (physical) layer.
+        if self._context.layers[self.vol.native_layer_name].is_valid(self.DeviceObject):
```

Comment:
It's a small nitpick, but I'd prefer we use the property self.context, rather than the underlying dataholder.  It gives us more flexibility in the future and prevents untrained developers from thinking they can access private members publicly...

Reply:
@ikelos We are confused by this comment. There is no `self.context` property - attempting to do that results in `AttributeError: StructType has no attribute: nt_symbols1!_FILE_OBJECT.context`. There are over 20 references to `self._context` among other extensions in the same `volatility/framework/symbols/windows/extensions/__init__.py ` file, but 0 references to `self.context`. Can you clarify please?

Reply:
Yep, sorry, `self.context` doesn't exist on objects, it's mostly for layers.  I could move it to `self.vol.context` except it's kinda different.  I'm not sure you'd ever need to externally ask an object for its context?  Given it's *self*, it's allowed to access protected variables, so I'm probably being overly sensitive.  Yep, completely ignore this, sorry!  5:S